Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -8,37 +11,53 @@ import { ArtifactItemStore } from '../Store/artifactItemStore';
 import { ArtifactEngineOptions } from "./artifactEngineOptions"
 import { Logger } from './logger';
 import { Worker } from './worker';
+import { TicketState } from '../Models/ticketState';
+import { CacheProvider } from '../Providers/cacheProvider';
 
 export class ArtifactEngine {
     processItems(sourceProvider: models.IArtifactProvider, destProvider: models.IArtifactProvider, artifactEngineOptions?: ArtifactEngineOptions): Promise<models.ArtifactDownloadTicket[]> {
         var artifactDownloadTicketsPromise = new Promise<models.ArtifactDownloadTicket[]>((resolve, reject) => {
             const workers: Promise<void>[] = [];
             artifactEngineOptions = artifactEngineOptions || new ArtifactEngineOptions();
+            artifactEngineOptions.artifactCacheDirectory = artifactEngineOptions.artifactCacheDirectory ? artifactEngineOptions.artifactCacheDirectory : tl.getVariable("AGENT_WORKFOLDER");
             this.createPatternList(artifactEngineOptions);
+            var artifactName = sourceProvider.getRootItemPath();
+            this.artifactItemStore = new ArtifactItemStore(artifactName);
             this.artifactItemStore.flush();
             Logger.verbose = artifactEngineOptions.verbose;
             this.logger = new Logger(this.artifactItemStore);
             this.logger.logProgress();
             sourceProvider.artifactItemStore = this.artifactItemStore;
             destProvider.artifactItemStore = this.artifactItemStore;
+            if(!artifactEngineOptions.enableIncrementalDownload) {
+                artifactEngineOptions.artifactCacheDirectory = '';
```

Comment:
why do we need to reset these keys? Is it a way to disable incremental download? doesn't look clean.

Reply:
No. we reset because if incremental download is not enabled then the `artifactCacheDirectory` and `artifactCacheKey` are not set and are returned undefined which throws an error while initializing the cacheProvider